ompi/info/info.c

```diff
@@ -116,76 +116,85 @@ int ompi_mpiinfo_init(void)
     return OMPI_SUCCESS;
 }
 
-/*
- * Fill in the MPI_INFO_ENV if using MPI3 initialization
+/**
+ * Fill in an info object with ENV info.  Used for setting
+ * MPI_INFO_ENV and by invocation of MPI_Info_create_env.
  */
-int ompi_mpiinfo_init_mpi3(void)
+
+int ompi_mpiinfo_init_env(int argc, char **argv, ompi_info_t *info)
 {
     char *cptr, **tmp;
 
     /* fill the env info object */
 
     /* command for this app_context */
-    if (NULL != ompi_process_info.command) {
+    if (NULL != argv) {
+        tmp = argv;
+    } else if (NULL != ompi_process_info.command) {
         tmp = opal_argv_split(ompi_process_info.command, ' ');
-        opal_info_set(&ompi_mpi_info_env.info.super, "command", tmp[0]);
+    }
+
+    if (NULL != tmp) {
+        opal_info_set(&info->super, "command", tmp[0]);
 
         /* space-separated list of argv for this command */
         if (1 < opal_argv_count(tmp)) {
             cptr = opal_argv_join(&tmp[1], ' ');
         } else {
             cptr = strdup(tmp[0]);
```

Comment:
Look at [this line](https://github.com/mpi4py/mpi4py-testing/runs/8251591711?check_suite_focus=true#step:17:18) of the mpi4py stack trace involving `strdup` being called inside [`ompi_mpiinfo_init_env`](https://github.com/mpi4py/mpi4py-testing/runs/8251591711?check_suite_focus=true#step:17:19).

That's the reason I mentioned above the possible issue with `tmp` initialization.

Reply:
tmp should be initialized to NULL.  adding to this PR.  Thanks for finding this!